src/platform/intel/ace/lib/clk.c

```diff
@@ -11,9 +11,26 @@
 #include <sof/common.h>
 #include <sof/lib/memory.h>
 #include <rtos/spinlock.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clk, LOG_LEVEL_DBG);
 
 static SHARED_DATA struct clock_info platform_clocks_info[NUM_CLOCKS];
 
+static int select_cpu_freq(int clock, int hz)
+{
+	const struct device *clock_ctrl = device_get_binding("clkctl");
+
+	if (!clock_ctrl) {
+		LOG_ERR("Could not find clock controller!");
+		return -ENODEV;
+	}
```

Comment:
and as it is compile time assignment you probably dont need the checks

Reply:
removed